src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5904,4 +5904,7 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="IDS_FeatureNestedStackalloc" xml:space="preserve">
     <value>stackalloc in nested expressions</value>
   </data>
+  <data name="ERR_NeedSpaceBetweenExclamationAndEquals" xml:space="preserve">
+    <value>There needs to be a space between the null-checking exclamation and the optional parameter's equals sign.</value>
```

Comment:
1. Consider simplifying to `space required between explanation-point and equals-sign here.`
2. the parser is happy to rip tokens apart if appropriate.  i.e. it can take `>>` and make it into two `>` tokens (or vice versa).  Should we consider doing tha there?  probably not important enough, but something thta could be considered.

Reply:
Regarding the message, perhaps "A space is required between the exclamation point and equals sign.". 

Regarding splitting the token, that sounds reasonable, but perhaps as a follow up work item.

Reply:
The error is added after the token is split.

Reply:
Why not use '!' and '=' instead of exclamation point and equals?

Reply:
@YairHalberstadt, using '!' and '=' in the message makes sense. So, perhaps the message should be: "A space is required between '!' and '='."